core/templates/pages/contributor-dashboard-page/modal-templates/translation-modal.component.ts

```diff
@@ -401,7 +401,41 @@ export class TranslationModalComponent {
     return this.translateTextService.getActiveIndex() > 0;
   }
 
-  skipActiveTranslation(): void {
+  getTrimmedTranslationContent(): string {
+    const content = Array.isArray(this.activeWrittenTranslation)
+      ? this.activeWrittenTranslation.join('')
+      : this.activeWrittenTranslation || '';
+    return content.trim();
+  }
+
+  confirmUnsavedChanges(isSaveDisabled: boolean): boolean {
+    if (isSaveDisabled) {
+      return true;
+    }
+    const translationContent = this.getTrimmedTranslationContent();
+    if (translationContent !== '') {
+      return window.confirm(
+        'You have unsaved changes. Do you want to discard them?'
+      );
+    }
+    return true;
+  }
+
+  isSaveButtonDisabled(): boolean {
```

Comment:
This seems to check a bunch of things, but not whether the save button is disabled or not.